### PR TITLE
Improvement: tracking type refactoring

### PR DESCRIPTION
**Pull request by Victor-Salomon**
This PR contains the **second** part of the transfer tracking cleaning & improvements: Types

Again, no big logic change.

- API route now supports WithinPolkadot transfer direction. 
- TxTrackingResult type provides more clarity within the code.

### CODE REVIEW

File: app/src/models/subscan.ts [hunk 1]
```diff
@@ -148,3 +65,89 @@ export type SubscanXCMTransferResult = {
   metadata: SubscanXCMMetadata
   status: TransferStatus
 }
+
+// LEGACY CODE BELOW
+// TO KEEP UNTIL FURTHER NOTICE
```
Comment (NunoAlexandre) [inline app/src/models/subscan.ts:70]: Why do we want to keep this commented code?

File: app/src/models/transfer.ts [hunk 2]
```diff
@@ -88,3 +87,5 @@ export enum TransferTab {
   Completed = 'Completed',
 }
 export type TransferTabOptions = TransferTab
+
+export type TxTrackingResult = FromEthTrackingRes | FromParachainTrackingRes
```
Comment (NunoAlexandre) [inline app/src/models/transfer.ts:91]: Can we be consistent with the usage of `Result` vs `Res`? 🙏

File: app/src/models/subscan.ts (hunk 1 above)
Comment (Victor-Salomon) [reply app/src/models/subscan.ts:70]: I want to keep it until we test Subscan endpoint support properly Parachain to AH transfers.

File: app/src/models/transfer.ts (hunk 2 above)
Comment (Victor-Salomon) [reply app/src/models/transfer.ts:91]: Obviously :)

File: app/src/models/subscan.ts (hunk 1 above)
Comment (noahjoeris) [reply app/src/models/subscan.ts:70]: Usually I would say it is in VCS anyways, but with Squash Merge I am actually not sure 😄

Comment (Victor-Salomon) [reply app/src/models/subscan.ts:70]: You're right 😁
Let me remove it.

**Review by noahjoeris — APPROVED** [review]
Great 🚀 Just the naming stuff Nuno mentioned above.

**Pull request merged**